vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -552,6 +557,13 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+        // Get the power state for the virtual machine.
+        if vprops.Runtime.PowerState == types.VirtualMachinePowerStatePoweredOn {
+                d.Set("power_state", "running")
+        } else {
+                d.Set("power_state", "shutdown")
+        }
```

Comment:
It would be nice to account for the "suspended" state here too, and perhaps use the state descriptions from the provider? Perhaps switch on `vprops.Runtime.PowerState`… something like:

```golang
switch vprops.Runtime.PowerState {
case types.VirtualMachinePowerStatePoweredOn:
    d.Set("power_state", "poweredOn")
case types.VirtualMachinePowerStatePoweredOff:
    d.Set("power_state", "poweredOff")
case types.VirtualMachinePowerStateSuspended:
    d.Set("power_state", "suspended")
}
```

Reply:
I would also agree that suspended should be included.

Reply:
I also agree that the names should reflect the following, _which are also seen in PowerCLI_:

- `PoweredOn`
- `PoweredOff`
- `Suspended`

Reply:
Resolved in https://github.com/hashicorp/terraform-provider-vsphere/pull/1407/commits/4925157671e97d52889181d977cad8123b92473d